frontend/src/page/Realm.tsx

```diff
@@ -1,43 +1,65 @@
 import React from "react";
 import { Link } from "react-router-dom";
 
+import { graphql, useLazyLoadQuery } from "react-relay/hooks";
+import { RealmQuery } from "../query-types/RealmQuery.graphql";
+
 import { Breadcrumbs } from "../ui/Breadcrumbs";
 
 
 type Props = {
-    path: string[];
+    path: string;
 };
 
 export const Realm: React.FC<Props> = ({ path }) => {
-    const isRoot = path.length === 0;
-    const ids = resolvePath(path);
-    if (ids == null) {
+    const isRoot = path === "";
+
+    // TODO Build this query from fragments!
+    const { realm } = useLazyLoadQuery<RealmQuery>(graphql`
+        query RealmQuery($path: String!) {
+            realm: realmByPath(path: $path) {
+                name
+                parents {
+                    name
+                    path
+                }
+                children {
+                    id
+                    name
+                    path
+                }
+            }
+        }
+    `, {
+        // The API expects a "leading" slash to non-root paths to separate
+        // the first component from the root path segment `''`
+        path: isRoot ? "" : `/${path}`,
```

Comment:
I... don't understand. This seems not so great.

Reply:
The full path of a realm is all the path segments from the root to it, joined by `"/"`.

The root realm has a path segment of `""`.

So the root has the full path `""` but every non-root realm has a full path starting with `"/"`.

I concede this seems questionable at first sight, but I think it's actually the most consistent pattern. Also, every other pattern would have an edge case like this in some other place, I think.

Reply:
Ah mh ok. Yeah makes more sense now. I guess we can't avoid some special casing somewhere.